server-api/src/main/java/org/apache/atlas/RequestContext.java

```diff
@@ -44,8 +44,9 @@
 public class RequestContext {
     private static final Logger METRICS = LoggerFactory.getLogger("METRICS");
 
-    private static final ThreadLocal<RequestContext> CURRENT_CONTEXT    = new ThreadLocal<>();
-    private static final Set<RequestContext>         ACTIVE_REQUESTS    = new HashSet<>();
+    private static final ThreadLocal<RequestContext> CURRENT_CONTEXT           = new ThreadLocal<>();
+    private static final Set<RequestContext>         ACTIVE_REQUESTS           = new HashSet<>();
+    private static final ThreadLocal<Boolean>        isUpdateNotification      = new ThreadLocal<>();
```

Comment:
Instead of having `isUpdateNotification` as a `static ThreadLocal<>`, have it as a instance member of `RequestContext` (similar to other members like `requestTime`). This will also make it unnecessary to reset the flag in NotificationHookConsumer.